test/test_strings.py

```diff
@@ -30,19 +30,12 @@ def _parse_test(self, filename):
         """Helper function to parse a binary file and check whether
         the given string is in the parsed result"""
         self.strings.filename = os.path.join(BINARIES_PATH, filename)
-        f = tempfile.TemporaryFile()
-        subprocess.call(["strings", self.strings.filename], stdout=f)
-        binutils_strings = f.readlines()
-        t1 = perf_counter()
-        ours = self.strings.parse().split("\n")
-        t2 = perf_counter()
-        print(f"pstring: {t2 - t1}")
-        t1 = perf_counter()
-        ours = self.strings.parse_3().split("\n")
-        t2 = perf_counter()
-        print(f"string: {t2 - t1}")
-        for theirs in binutils_strings:
-            self.assertIn(theirs.decode("utf-8"), ours)
+        with tempfile.TemporaryFile() as f:
+            subprocess.call(["strings", self.strings.filename], stdout=f)
+            binutils_strings = f.readlines()
```

Comment:
Have you made sure that this is really getting all the lines? Usually we need a `f.seek(0)` before a read and after it was written to.

Reply:
Yeah I have put f.seek(0) now and fixed sum bugs.
```console
New results:
subprocess strings: 0.001479687000028207
pure python: 0.0007050880003589555
```